### PR TITLE
Remove references to baseBranch

### PR DESCRIPTION
**Pull request by mehulkar**

### CODE REVIEW

File: docs/pages/docs/getting-started.mdx [hunk 1]
```diff
@@ -109,7 +109,7 @@ The `turbo` package is a little shell that will install the proper `turbo-<os>-<
 
 In the root of your monorepo, create an empty file named `turbo.json`. This will hold the configuration for Turborepo.
 
-If your git repo's base branch is not `origin/master`, then you need to specify a `baseBranch` too. For example, `origin/main`.
+If your git repo's base branch is not `origin/main`, then you need to specify a `baseBranch` too. For example, `origin/main`.
```
Comment (chris-olszewski) [inline docs/pages/docs/getting-started.mdx:112]: The example should be updated to show non default behavior now that `baseBranch` defaults to `origin/main`

Comment (mehulkar) [reply docs/pages/docs/getting-started.mdx:112]: 🤦🏾  good catch, updated

**Review by chris-olszewski — DISMISSED** [review]
Trying to trace through this, I'm not seeing us actually default to `origin/master` or `origin/main` anywhere. I could easily be missing something, but if this is the case then we shouldn't be claiming a default.

**Comment by mehulkar** [comment]
> Trying to trace through this, I'm not seeing us actually default to `origin/master` or `origin/main` anywhere. I could easily be missing something, but if this is the case then we shouldn't be claiming a default.

Good call. I don't see it being used either. @gsoltis @tknickman @nathanhammond do you know if this config is stil being used? I see it's in the type def:

https://github.com/vercel/turborepo/blob/3837c41bc313d8a9a648dcb0e05bb498541fe234/cli/internal/fs/turbo_json.go#L16

but I don't see any uses of `.Base` property where `ReadTurboJSON()` is called

**Comment by gsoltis** [comment]
you're correct, we're not using it. git comparisons are either done against the current `HEAD` or specified inline in the value passed to `--filter`

**Comment by mehulkar** [comment]
Looks like it's been there since the original commit in this history: 6077c5e4894f4fa0a95b01d5cc8f3f103a227b8b, and even at that commit, I don't see it being used.

I can try going through and ripping all this out, but is there any reason to think this is a breaking change? The first commit in this repo is pre 1.0, so I _think_ we can assume that the `baseBranch` implementation (if there ever was one) has not worked since we hit 1.0.

cc @jaredpalmer if you have historical context or any other advice here!

**Comment by jaredpalmer** [comment]
Should be removed. Was used in JS version pre-acquisition and pre-Go and I guess never removed.

**Review by gsoltis — APPROVED** [review]

**Pull request merged**